### PR TITLE
[KYUUBI #5435][INFRA][TEST] Improve Kyuubi On Kubernetes IT

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_

Improve Kyuubi On Kubernetes IT

Done:
1. Copy spark submit engine log in kyuubi pod to local and upload when test failed.
2. pre install spark image into minikube to avoid image pull error

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

No

### CODE REVIEW

**Comment by zwangsheng** [comment]
@wForget @pan3793 PTAL

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -403,6 +403,9 @@ jobs:
           minikube start --cpus 2 --memory 4096 --kubernetes-version=${KUBERNETES_VERSION} --force
           # https://minikube.sigs.k8s.io/docs/handbook/pushing/#7-loading-directly-to-in-cluster-container-runtime
           minikube image load apache/kyuubi:latest
+          # pre-install spark into minikube
+          docker pull apache/spark:3.4.1
```
Comment (pan3793) [inline .github/workflows/master.yml:407]: cachable?

Comment (zwangsheng) [reply .github/workflows/master.yml:407]: I can't found how to cache pulled docker image.

Comment (pan3793) [reply .github/workflows/master.yml:407]: then how does it help? just perform the pulling action slightly in advance?

Comment (pan3793) [reply .github/workflows/master.yml:407]: alright, we educated that `kubectl docker` has an unstable network even though we don't know why.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Merged to master/1.8

**Pull request closed without merging**